src/spyglass/spikesorting/v0/spikesorting_recording.py

```diff
@@ -332,37 +338,94 @@ def make(self, key):
         2. Saves the recording data to the recording directory
         3. Inserts the path to the recording data into SpikeSortingRecording
         """
-        sort_interval_valid_times = self._get_sort_interval_valid_times(key)
-        recording = self._get_filtered_recording(key)
-        recording_name = self._get_recording_name(key)
-
-        # Path to files that will hold the recording extractors
-        recording_path = str(recording_dir / Path(recording_name))
-        if os.path.exists(recording_path):
-            shutil.rmtree(recording_path)
-
-        recording.save(
-            folder=recording_path, chunk_duration="10000ms", n_jobs=8
-        )
+        rec_info = self._make_file(key)
 
         IntervalList.insert1(
             {
                 "nwb_file_name": key["nwb_file_name"],
-                "interval_list_name": recording_name,
-                "valid_times": sort_interval_valid_times,
+                "interval_list_name": rec_info["name"],
+                "valid_times": self._get_sort_interval_valid_times(key),
                 "pipeline": "spikesorting_recording_v0",
             },
             replace=True,
         )
 
-        self.insert1(
-            {
-                **key,
-                # store the list of valid times for the sort
-                "sort_interval_list_name": recording_name,
-                "recording_path": recording_path,
-            }
+        self_insert = dict(
+            key,
+            sort_interval_list_name=rec_info["name"],
+            recording_path=rec_info["path"],
+        )
+        self.insert1(self_insert)
+        self._record_environment(self_insert)
+
+    def _record_environment(self, key):
+        """Record environment details for this recording."""
+        from spyglass.spikesorting.v0 import spikesorting_recompute as rcp
+
+        rcp.RecordingRecomputeSelection.insert(key, at_creation=True)
+
+    def _make_file(self, key, base_dir=None, return_hasher=False):
+        """Run only operations required to save the recording data to disk."""
+        has_entry = bool(self & key)  # table entry exists, so recompute files
+        base_dir = Path(base_dir or recording_dir)
+        rec_path = base_dir / Path(self._get_recording_name(key))
+        ret = {"name": self._get_recording_name(key), "path": str(rec_path)}
+
+        if rec_path.exists():
+            if has_entry:  # if table entry for existing file, use it
+                return {**ret, "hash": self._dir_hash(rec_path)}
+            else:  # if no table entry, assume existing is outdated and delete
+                shutil.rmtree(rec_path)
```

Comment:
Is there any consequence of failing to remove the file?

Reply:
As in a consequence of this `rmtree` failing? It would raise an error, which would prevent the rest of the func from running. It might be the case that a subset of files remain. The user can either...
- Address the error and rerun
- Attempt to rerun and hit the same error
I don't think anything prior to this point changes the database or filesystem state

Reply:
And rerunning with a subset of files won't have any consequences?

Reply:
It looks like the only way to tread `rmtree` as a transaction is to run a backup first, then delete. That would add a lot of time for copying big files, so I'm adding a check that the folder has at least the appropriate number of files before considering the v0 folder 'present'

Is it sufficient to check the number is at least the typical file count? This won't catch all file tampering (remove N, add N+1), but it seems to protect against the interrupted `rmtree` leaving things behind